src/main/scala/chiseltest/backends/SimulatorInterface.scala

```diff
@@ -0,0 +1,30 @@
+package chiseltest.backends
+
+import firrtl.AnnotationSeq
+
+/** API to Simulator. */
+trait SimulatorInterface {
+
+  /** start time of this test. */
+  final val startTime: Long = System.nanoTime()
+
+  /** how many nano seconds has elapsed since this test start. */
+  def elapsedNanoSeconds(): Long = System.nanoTime() - startTime
+
+  def annotations: AnnotationSeq
+
+  /** poke a signal name with a [[BigInt]]. */
+  def poke(signal: String, value: BigInt): Unit
+
+  /** peek a signal name, return a BigInt if found. */
+  def peek(signal: String): Option[BigInt]
+
+  /** step the main clock. */
+  def step(n: Int): Unit
+
+  /** end simulator. */
+  def finish(): Unit
+
+  /** interface to get current clock. */
+  def cycleCount: BigInt
```

Comment:
Cycle count on what? What if there are multiple clocks? What is the top-level isn't a module with an implicit clock?

Reply:
this API should be removed.